src/main/java/org/assertj/core/internal/Iterables.java

```diff
@@ -1147,6 +1149,33 @@ public void assertContainsExactly(AssertionInfo info, Iterable<?> actual, Object
       throw failures.failure(info, elementsShouldSatisfy(actual, unsatisfiedRequirements, info));
   }
 
+  private <E> Function<? super Consumer<? super E>, List<E>> listFilteredBySatisfiedElements(Iterable<? extends E> actual) {
```

Comment:
can be static

Reply:
name is confusing, I was expecting a list from this method

Reply:
You are right, I renamed it.